src/routes/PrivateRoute.tsx

```diff
@@ -0,0 +1,24 @@
+// https://v5.reactrouter.com/web/example/route-config
+// https://github.com/remix-run/react-router/blob/c13b66939ef48eacf7067f7aec4752777be8b17c/docs/api-reference.md
+// Here is a different implementation of PrivateRoute using Outlet
+
+import { useAppSelector } from "../app/hook";
+import { userStateSelector } from "../features/user/userSlice";
+import { Navigate } from "react-router-dom";
+
+type IRouteProps = {
+  ComponentProp: any
+}
+
+const PrivateRoute = ({ ComponentProp }: IRouteProps): JSX.Element => {
```

Comment:
```suggestion
const PrivateRoute = ({ component}: IRouteProps) => {
```
Just `component`. CamelCase style should not be used in prop. No necessary define return type in this case